src/ast/passes/semantic_analyser.cpp

```diff
@@ -1132,6 +1137,26 @@ void SemanticAnalyser::visit(Call &call)
 
     call.type = CreateUInt64();
   }
+  else if (call.func == "offsetof")
+  {
+    check_nargs(call, 2);
+    auto &element_arg = *call.vargs->at(1);
+    String &element = static_cast<String&>(element_arg);
+    auto &arg = *call.vargs->at(0);
+    // check_arg() cannot be applied to offsetof(), because
+    // Type::record is first arg of offsetof(). check_arg()
+    // return true anyway.
+    if (arg.type.type != Type::record) {
+      LOG(ERROR, call.loc, err_)
+          << call.func << "() only supports " << Type::record << " arguments ("
+          << arg.type.type << " provided)";
+    }
+    // Only case Type::record can call HasField().
+    else if (!call.vargs->at(0)->type.HasField(element.str))
+      LOG(ERROR, call.loc, err_)
+          << "struct has no this field";
```

Comment:
```suggestion
          << "struct \'" << arg.type.GetName() << "\' has no field \'" 
          << ... << "\'";
```
Let's make the error messages as informative as possible.